Assets/Scripts/SingleControlFlowInstruction.cs

```diff
@@ -7,12 +7,26 @@ public abstract class SingleControlFlowInstruction : ControlFlowInstruction {
         public override int GetNumArguments() {
             return 3;
         }
-        protected override StandAloneInstruction GetExitInstruction() {
+        protected override StandAloneInstruction GetNestedInstruction() { 
+            return GetArgumentAt(0) as StandAloneInstruction;
+        }
+        protected override string GetNestedInstructionsAsString(){
+            string result = "";
+            StandAloneInstruction currInstruction = GetNestedInstruction();
+            while (currInstruction != null){
+                result += "\n\t" + currInstruction.DescriptiveInstructionToString().Replace("\n\t", "\n\t\t"); //add nested instructions with accumulated tabbing
+                currInstruction = currInstruction.GetNextInstruction();
+            }
+            return result;
+
+        }
+
+        public override StandAloneInstruction GetNextInstruction(){ //overriding Next
             return GetArgumentAt(2) as StandAloneInstruction;
         }
 
         public override string DescriptiveInstructionToString() {
-            return string.Join("", ToString(), GetArgumentAt(1)?.DescriptiveInstructionToString(), " then ", GetArgumentAt(0)?.DescriptiveInstructionToString());
+            return string.Join("", "<color=purple>", ToString(), "</color>", GetArgumentAt(1)?.DescriptiveInstructionToString(), ": ", GetNestedInstructionsAsString());
```

Comment:
any change you could make a script similar to `ResourcePathConstants.cs` but for colors? Or does something for Unity exist already for these strings?

Reply:
hmm yes I could, though it may be excessive if we're only going to have purple for keywords and white otherwise?